src/controllers/portfolio/portfolio.ts

```diff
@@ -793,13 +793,14 @@ export class PortfolioController extends EventEmitter {
               portfolioLib,
               {
                 blockTag: 'pending',
-                ...(currentAccountOps && {
-                  simulation: {
-                    account: selectedAccount,
-                    accountOps: currentAccountOps
-                  }
-                }),
-                isEOA: !isSmartAccount(selectedAccount),
+                ...(currentAccountOps &&
+                  state && {
```

Comment:
Why do we need to check for `state` (AccountOnchainState) here?  When we run a simulation, isn’t the state already fetched and defined in earlier steps?

Reply:
You are right, it is defined in earlier steps.  
I see this more as a precaution of a rare case where the RPC doesn't work for some reason and returns an invalid/empty state for `AccountOnChainState`. There's risk an error is thrown here instead of just skipping the simulation which would be the lesser evil in my opinion

Reply:
On the other hand, when we do the simulation, we don't use this passed `state: AccountOnchainState` anywhere. If it is not used in the portfolio, how this will reflect the simulation and may result in an error?

If you have a strong arguments that we need to do a strong validation for state existance, we could do it in the 2 places we call `this.portfolio.updateSelectedAccount()` in main ctrl , instead of passing the whole state and not using it in the logic. But would be nice to include how the missing state could break the simulation.